packages/react-core/src/components/Timestamp/Timestamp.tsx

```diff
@@ -0,0 +1,127 @@
+   */
+  hasUTCTooltip?: boolean;
+  /** Flag for displaying the time in a 12 hour format (true) or a 24 hour format (false),
+   * when custom content is not passed in.
+   *
+   * If this prop is not passed in, the hour format will be based on the locale prop's value. */
+  is12Hour?: boolean;
+  /** Determines which locale to use in the displayed content when custom content is not
+   * passed in. Defaults to the current locale when this prop is not passed in.
+   * For more information about locale's:
+   * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl#Locale_identification_and_negotiation
+   */
+  locale?: string;
+  /** Determines the format of the displayed time when custom content is not passed in. Examples:
+   * "Full" => 11:25:00 AM Eastern Daylight Time
+   * "Long" => 11:25:00 AM EDT
+   * "Medium" => 11:25:00 AM
+   * "Short" => 11:25 AM
+   */
+  timeFormat?: 'full' | 'long' | 'medium' | 'short';
+  /** Applies a custom suffix to the displayed time when custom content is not passed in.
+   * This prop should only be passed in when the timeFormat prop has a value of "medium" or "short".
+   */
+  timeZoneSuffix?: string;
+  /** Adds a custom tooltip to the timestamp. */
+  tooltipProps?: TooltipProps;
+  /** Applies a custom suffix to the UTC tooltip. */
+  utcSuffix?: string;
+}
+
+export const Timestamp: React.FunctionComponent<TimestampProps> = ({
+  children,
+  className,
+  customFormat,
+  date: dateProp = new Date(),
+  dateFormat,
+  datetime,
+  hasUTCTooltip,
+  is12Hour,
+  locale = undefined,
+  timeFormat,
+  timeZoneSuffix = '',
+  tooltipProps,
+  utcSuffix = 'UTC',
+  ...props
+}: TimestampProps) => {
+  const hasTooltip = hasUTCTooltip || tooltipProps?.content;
+  const formatOptions = {
+    ...(dateFormat && !customFormat && { dateStyle: dateFormat }),
+    ...(customFormat && { ...customFormat }),
+    ...(is12Hour !== undefined && { hour12: is12Hour })
+  };
+  const hasTimeFormat = timeFormat && !customFormat;
+
+  const dateAsLocaleString = new Date(dateProp).toLocaleString(locale, {
+    ...formatOptions,
+    ...(hasTimeFormat && { timeStyle: timeFormat })
+  });
+
+  const utcTimeFormat = timeFormat !== 'short' ? 'medium' : 'short';
+  const convertToUTCString = (date: Date) => new Date(date).toUTCString().slice(0, -3);
+  const utcDateString = new Date(convertToUTCString(dateProp)).toLocaleString(locale, {
+    ...formatOptions,
+    ...(hasTimeFormat && { timeStyle: utcTimeFormat })
+  });
+
+  const timestamp = (
+    <span
+      className={css(styles.timestamp, hasTooltip && styles.modifiers.helpText, className)}
+      {...(hasTooltip && { tabIndex: 0 })}
+      {...props}
+    >
+      <time dateTime={datetime}>{!children ? `${dateAsLocaleString} ${timeZoneSuffix}` : children}</time>
+    </span>
+  );
+
+  return hasTooltip ? (
+    <Tooltip content={tooltipProps?.content || `${utcDateString} ${utcSuffix}`} {...tooltipProps}>
```

Comment:
I do not see an example of the tooltip in core.  I do however see a mockup in the design. The design mentions a Popover though.
I think we should use a similar API here as we do for the Progress stepper help text where we use the render prop and pass in the popover.

Reply:
I just saw conversation in issue.  SO it should be a tooltip.  I would still suggest a similar API to the progress stepper's help text.

Reply:
Originally that was what I was considering, but I ended up going with this approach to avoid using refs since there was previously an issue with tooltips via refs getting announced correctly (though tooltips with refs do work better now). I can update to be more similar to how the Progress Stepper uses Popover's. 

Based on conversation in #7729 , I had opted to have one prop for rendering a default, UTC tooltip, and another for rendering a custom tooltip. Do you feel having both would still be good (replacing the current `tooltipProps` prop with a `tooltipRender` prop that matches ProgressStepper)?

Reply:
I think it is best to be consistent if we can.  If the we truly can't make the tooltip accessible with the API we use elsewhere for help text, then we can consider something else.
That being said, since we want a  default tooltip here.  I still think we can improve the API a bit.  I think using `tooltipProps` for custom tooltips is little confusing.  Maybe we can combine the props 

```
export interface TimestampTooltipObject {
  /** variant of tooltip*/
  variant: 'default' | 'custom' ;
  /** custom content */
  content?: React.ReactNode;
 /** Additional tooltip props **/
  tooltipProps?: TooltipProps;
}
```
Or something like.  Then if the object exists you can render a tooltip.